centaur/src/main/scala/centaur/json/JsonUtils.scala

```diff
@@ -71,9 +91,13 @@ object JsonUtils {
     def nonEmptyObjectArray = jsArray.isObjectArray && jsArray.nonEmpty
     def isSingleCallArray = jsArray.hasShardIndex && jsArray.size == 1
 
-    def hasShardIndex: Boolean = {
-      if (jsArray.nonEmptyObjectArray) jsArray.elements.map(_.asJsObject) forall { _.hasShardIndex }
+    def hasField(f: JsObject => Boolean): Boolean = {
```

Comment:
I think I would change `EnhancedJsObject` to have a `hasField(fieldName: String): Boolean` method to replace its current `hasShardIndex`. Then this method could just be `def hasField(fieldName: String): Boolean` too and might look easier to work out what it's doing?

Reply:
That is certainly more cleaner. I have updated the code.